package/tests/test_data_access/test_managers.py

```diff
@@ -336,6 +340,30 @@ def test_add_pipelines_with_transcoded_data(
             for node in data_access_manager.nodes.as_list()
         )
 
+    def test_different_reigstered_pipelines_having_modular_pipeline_with_same_name(
+        self,
+        data_access_manager: DataAccessManager,
+    ):
+        # this test case was taken from the following user's report:
+        # https://github.com/kedro-org/kedro-viz/issues/858
+        registered_pipelines = {
+            "__default__": pipeline(
+                [node(func=lambda a: False, inputs="tst.a", outputs="d")]
+            ),
+            "pipe2": pipeline(
+                [node(func=lambda a: False, inputs="tst.b", outputs="tst.c")]
+            ),
+        }
+
+        data_access_manager.add_catalog(DataCatalog())
+        data_access_manager.add_pipelines(registered_pipelines)
+        modular_pipeline_tree = (
+            data_access_manager.create_modular_pipelines_tree_for_registered_pipeline(
+                DEFAULT_REGISTERED_PIPELINE_ID
+            )
+        )
+        assert len(modular_pipeline_tree["tst"].children) == 1
```

Comment:
Without the fix in this PR, this number will be `3` because it mistook `tst.b` and `tst.c` as part of the `__default__` pipeline which is what the user reported.